app/src/main/java/com/vero/tasky/MainActivity.kt

```diff
@@ -7,20 +7,30 @@ import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Surface
 import androidx.compose.ui.Modifier
+import androidx.navigation.compose.rememberNavController
+import com.vero.tasky.core.domain.local.Preferences
+import com.vero.tasky.core.presentation.components.RootNavigation
 import com.vero.tasky.ui.theme.TaskyTheme
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
+
+    @Inject
+    lateinit var preferences: Preferences
```

Comment:
Personally I have a MainViewModel, where I handle this

Reply:
Yes, you rarely need to really inject something in an Activity other than a ViewModel. I'd keep the UI as stupid as possible and leave stuff like this up to the ViewModel which makes it also easier to test this behavior (otherwise you'd need to setup a UI test)